core/src/main/scala/kafka/cluster/Partition.scala

```diff
@@ -285,6 +314,17 @@ class Partition(val topicPartition: TopicPartition,
   // If ReplicaAlterLogDir command is in progress, this is future location of the log
   @volatile var futureLog: Option[UnifiedLog] = None
 
+  // Partition listeners
+  private val listeners = new CopyOnWriteArrayList[PartitionListener]()
+
+  private val logOffsetsListener = new LogOffsetsListener {
+    override def onHighWatermarkUpdated(offset: Long): Unit = {
```

Comment:
For my understanding, they way we update the Partition Listeners is that we have our own Log layer listener we rely on?

Reply:
> For my understanding, they way we update the Partition Listeners is that we have our own Log layer listener we rely on for this purpose?

That's correct.

> Will we use the logOffsetsListener for any other purposes?

Not at the moment.